### PR TITLE
Proposal to add a hard_close method to TCPConnection

### PR DESCRIPTION
**Pull request by joncfoo**

### CODE REVIEW

File: text/0000-tcpconnection-hard-close.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+possible to hard close the connection as `TCPConnection.close()` only performs a
+hard close if the connection is muted. Muting the connection however is only
+available via the `TCPConnection.mute()` behavior that is run asynchronously.
+Hence one cannot call `conn.mute()` followed by `conn.close()` in order to hard
+close the connection.
+
+# Detailed design
+
+A new public method `fun ref hard_close()` would be created on `TCPConnection`.
+This method would invoke the existing private method `fun ref _hard_close()`
```
Comment (jemc) [inline text/0000-tcpconnection-hard-close.md]: Why not make the existing method public instead?

**Comment by SeanTAllen** [comment]
With the change of this to Joe's proposal to make _hard_close public, we are in favor of this. Once that change has been made. We'd be happy to accept. Pending further comments.

File: text/0000-tcpconnection-hard-close.md (hunk 1 above)
Comment (joncfoo) [reply text/0000-tcpconnection-hard-close.md]: Updated to make the existing method public.

**Comment by SeanTAllen** [comment]
@Theodus do you have time to update the bookkeeping on this, create the issue etc?

**Pull request merged**